### PR TITLE
rename config option .url and .ca to .hosts and .certificate_authorit

### PR DESCRIPTION
**Pull request by colinsurprenant**
Per the recent Kibana options changes in kibana#29496 we are also changing this options for 7.0.0-beta1.

- `*.elasticsearch.url` to `*.elasticsearch.hosts`
- `*.elasticsearch.ca` to `*.elasticsearch.certificate_authority`

This PR has 3 commits; one for the code changes, tests and docs.

### CODE REVIEW

File: x-pack/lib/config_management/extension.rb [hunk 1]
```diff
@@ -26,8 +26,8 @@ def additionals_settings(settings)
         settings.register(LogStash::Setting::ArrayCoercible.new("xpack.management.pipeline.id", String, ["main"]))
         settings.register(LogStash::Setting::NullableString.new("xpack.management.elasticsearch.username", "logstash_system"))
         settings.register(LogStash::Setting::NullableString.new("xpack.management.elasticsearch.password"))
-        settings.register(LogStash::Setting::ArrayCoercible.new("xpack.management.elasticsearch.url", String, [ "https://localhost:9200" ] ))
```
Comment (yaauie) [inline x-pack/lib/config_management/extension.rb]: will we need a backport in which we merely deprecate the current behaviour for 6.x (6.7, which is cut tomorrow)

**Comment by colinsurprenant** [comment]
@jsvd WDYT for the deprecation?

**Comment by colinsurprenant** [comment]
rebasing and merging.

**Comment by colinsurprenant** [comment]
unrelated failure #10371 - merging.

**Pull request merged**